File: frontend/src/constants/teams.js
export const TEAM_COLORS = {
  'Chennai Super Kings': { primary: '#FCCA06', secondary: '#0081E9', abbr: 'CSK' },
  'Mumbai Indians': { primary: '#2E8BF0', secondary: '#D1AB3E', abbr: 'MI' },
  'Royal Challengers Bangalore': { primary: '#EC1C24', secondary: '#2B2A29', abbr: 'RCB' },
  'Royal Challengers Bengaluru': { primary: '#EC1C24', secondary: '#2B2A29', abbr: 'RCB' },
  'Kolkata Knight Riders': { primary: '#7B5EA7', secondary: '#B3A123', abbr: 'KKR' },
  'Delhi Capitals': { primary: '#1768AC', secondary: '#EF1B23', abbr: 'DC' },
  'Delhi Daredevils': { primary: '#1768AC', secondary: '#EF1B23', abbr: 'DD' },
  'Punjab Kings': { primary: '#D4213D', secondary: '#A7A9AC', abbr: 'PBKS' },
  'Kings XI Punjab': { primary: '#D4213D', secondary: '#A7A9AC', abbr: 'KXIP' },
  'Rajasthan Royals': { primary: '#EA1A85', secondary: '#254AA5', abbr: 'RR' },
  'Sunrisers Hyderabad': { primary: '#FF822A', secondary: '#000000', abbr: 'SRH' },
  'Gujarat Titans': { primary: '#A7D8DE', secondary: '#1C1C2B', abbr: 'GT' },
  'Lucknow Super Giants': { primary: '#A72056', secondary: '#FFCC00', abbr: 'LSG' },
  'Deccan Chargers': { primary: '#C0C0CC', secondary: '#A7A9AC', abbr: 'DC' },
  'Rising Pune Supergiant': { primary: '#6F61AC', secondary: '#D63D70', abbr: 'RPS' },
  'Rising Pune Supergiants': { primary: '#6F61AC', secondary: '#D63D70', abbr: 'RPS' },
  'Gujarat Lions': { primary: '#E04F17', secondary: '#1C1C2B', abbr: 'GL' },
  'Pune Warriors': { primary: '#2F9BE3', secondary: '#E55B25', abbr: 'PWI' },
  'Kochi Tuskers Kerala': { primary: '#6F2C91', secondary: '#F7B731', abbr: 'KTK' },
}

export function getTeamColor(teamName) {
  return TEAM_COLORS[teamName]?.primary || '#8888A0'
}

export function getTeamAbbr(teamName) {
  return TEAM_COLORS[teamName]?.abbr || teamName?.substring(0, 3).toUpperCase() || '???'
}

// Team logo image filenames (stored in backend/team_images/)
const TEAM_LOGO_FILES = {
  'Chennai Super Kings': 'CSK.jpg',
  'Mumbai Indians': 'MI.jpg',
  'Royal Challengers Bangalore': 'RCB.jpg',
  'Royal Challengers Bengaluru': 'RCB.jpg',
  'Kolkata Knight Riders': 'KKR.png',
  'Delhi Capitals': 'DC.png',
  'Delhi Daredevils': 'DC.png',
  'Punjab Kings': 'PK.jpg',
  'Kings XI Punjab': 'PK.jpg',
  'Rajasthan Royals': 'RR.png',
  'Sunrisers Hyderabad': 'SRH.jpg',
  'Gujarat Titans': 'GT.png',
  'Lucknow Super Giants': 'LSG.png',
  'Deccan Chargers': 'Decaan.jpg',
  'Rising Pune Supergiant': 'RPSG.jpg',
  'Rising Pune Supergiants': 'RPSG.jpg',
  'Gujarat Lions': 'GL.jpg',
  'Pune Warriors': 'PW.jpg',
  'Kochi Tuskers Kerala': 'KT.png',
}

const API_BASE = import.meta.env.VITE_API_URL || ''

export function getTeamLogo(teamName) {
  const file = TEAM_LOGO_FILES[teamName]
  if (!file) return null
  return `${API_BASE}/api/team-images/${file}`
}

File: mobile/lib/core/teams.dart
import 'package:flutter/material.dart';
import 'config.dart';

class TeamInfo {
  final Color primary;
  final Color secondary;
  final String abbr;
  final String? logoFile;

  const TeamInfo(this.primary, this.secondary, this.abbr, [this.logoFile]);
}

const Map<String, TeamInfo> kTeamInfo = {
  'Chennai Super Kings': TeamInfo(
    Color(0xFFFCCA06),
    Color(0xFF0081E9),
    'CSK',
    'CSK.jpg',
  ),
  'Mumbai Indians': TeamInfo(
    Color(0xFF2E8BF0),
    Color(0xFFD1AB3E),
    'MI',
    'MI.jpg',
  ),
  'Royal Challengers Bangalore': TeamInfo(
    Color(0xFFEC1C24),
    Color(0xFF2B2A29),
    'RCB',
    'RCB.jpg',
  ),
  'Royal Challengers Bengaluru': TeamInfo(
    Color(0xFFEC1C24),
    Color(0xFF2B2A29),
    'RCB',
    'RCB.jpg',
  ),
  'Kolkata Knight Riders': TeamInfo(
    Color(0xFF7B5EA7),
    Color(0xFFB3A123),
    'KKR',
    'KKR.png',
  ),
  'Delhi Capitals': TeamInfo(
    Color(0xFF1768AC),
    Color(0xFFEF1B23),
    'DC',
    'DC.png',
  ),
  'Delhi Daredevils': TeamInfo(
    Color(0xFF1768AC),
    Color(0xFFEF1B23),
    'DD',
    'DC.png',
  ),
  'Punjab Kings': TeamInfo(
    Color(0xFFD4213D),
    Color(0xFFA7A9AC),
    'PBKS',
    'PK.jpg',
  ),
  'Kings XI Punjab': TeamInfo(
    Color(0xFFD4213D),
    Color(0xFFA7A9AC),
    'KXIP',
    'PK.jpg',
  ),
  'Rajasthan Royals': TeamInfo(
    Color(0xFFEA1A85),
    Color(0xFF254AA5),
    'RR',
    'RR.png',
  ),
  'Sunrisers Hyderabad': TeamInfo(
    Color(0xFFFF822A),
    Color(0xFF000000),
    'SRH',
    'SRH.jpg',
  ),
  'Gujarat Titans': TeamInfo(
    Color(0xFFA7D8DE),
    Color(0xFF1C1C2B),
    'GT',
    'GT.png',
  ),
  'Lucknow Super Giants': TeamInfo(
    Color(0xFFA72056),
    Color(0xFFFFCC00),
    'LSG',
    'LSG.png',
  ),
  'Deccan Chargers': TeamInfo(
    Color(0xFFC0C0CC),
    Color(0xFFA7A9AC),
    'DCH',
    'Decaan.jpg',
  ),
  'Rising Pune Supergiant': TeamInfo(
    Color(0xFF6F61AC),
    Color(0xFFD63D70),
    'RPS',
    'RPSG.jpg',
  ),
  'Rising Pune Supergiants': TeamInfo(
    Color(0xFF6F61AC),
    Color(0xFFD63D70),
    'RPS',
    'RPSG.jpg',
  ),
  'Gujarat Lions': TeamInfo(
    Color(0xFFE04F17),
    Color(0xFF1C1C2B),
    'GL',
    'GL.jpg',
  ),
  'Pune Warriors': TeamInfo(
    Color(0xFF2F9BE3),
    Color(0xFFE55B25),
    'PWI',
    'PW.jpg',
  ),
  'Kochi Tuskers Kerala': TeamInfo(
    Color(0xFF6F2C91),
    Color(0xFFF7B731),
    'KTK',
    'KT.png',
  ),
};

Color teamColor(String? name) =>
    kTeamInfo[name]?.primary ?? const Color(0xFF8888A0);

String teamAbbr(String? name) =>
    kTeamInfo[name]?.abbr ??
    (name == null || name.isEmpty
        ? '???'
        : name.substring(0, name.length.clamp(0, 3)).toUpperCase());

String? teamLogoUrl(String? name) {
  final file = kTeamInfo[name]?.logoFile;
  if (file == null) return null;
  return AppConfig.teamImage(file);
}
